src/Listener/SaveTrelloIdToDatabase.php

```diff
@@ -121,45 +120,28 @@ private function attachLabelsToCardBasedOnForumTags(Discussion $discussion, Card
             $board = (new Board($client))->setId($shortLink)->get();
 
             if ($board) {
-                $labels = collect($board->getLabels());
+                $mappings = json_decode($this->settings->get('blomstra-trello.label-tag-mappings'), true);
 
-                $includeSecondaryTags = (bool) $this->settings->get('blomstra-trello.include_secondary_tags_as_trello_labels', false);
+                $boardMappings = Arr::get($mappings, $shortLink);
 
-                $discussion->tags->filter(function ($tag) use ($includeSecondaryTags) {
-                    return $includeSecondaryTags ?: !is_null($tag->position);
-                })->each(function ($tag) use ($labels, $board, $card) {
-                    $label = $labels->filter(function ($label) use ($tag) {
-                        return $label->name == $tag->name;
-                    })->first();
+                if ($boardMappings) {
+                    $discussion->tags->each(function ($tag) use ($client, $boardMappings, $card) {
+                        foreach ($boardMappings as $boardMapping) {
+                            if ($boardMapping['tagId'] == $tag->id) {
+                                $labelId = data_get($boardMapping, 'label.id');
 
-                    if (!$label) {
-                        $label = $labels->filter(function ($label) {
-                            return $label->name == '';
-                        })->first();
+                                $label = new Label($client);
```

Comment:
Related to my above comment, I think we should not be passing the `$client` around here, and let the `Label` class resolve the client required

Reply:
@imorland can you check my latest code? I've already created a ServiceProvider and created a singleton for the client. Added it as a parameter in the __construct and store it to $this->client, but newing up a Label here still throws too few arguments exception if I don't pass a client.

Reply:
I've not looked at the changes yet, but if you've now got `Client` as a singleton, you don't need to pass it in a constructor now. Instead, `resolve(Client::class)` will give you that instance wherever you need it

Reply:
Right. I can resolve it already using it like that. I'll change it then by removing it in the construct.

Reply:
Still passing `$client`